runtime/bindings/python/hal.cc

```diff
@@ -331,7 +331,7 @@ HalDevice HalDriver::CreateDevice(iree_hal_device_id_t device_id) {
     // {"device_id": obj, "path": str, "name": str}.
     auto record_dict = py::cast<py::dict>(record);
     py::object found_device_id = record_dict["device_id"];
-    if (found_device_id.is(compare_device_id)) {
+    if (found_device_id.equal(compare_device_id)) {
```

Comment:
Digging through the history here, I'm wondering why was this broken... are we missing a unit test somewhere?

https://github.com/iree-org/iree/pull/9330 added this as
```c
    if (found_device_id == compare_device_id) {
      found = true;
```
then https://github.com/iree-org/iree/pull/10899 changed it to
```diff
-    if (found_device_id == compare_device_id) {
+    if (found_device_id.is(compare_device_id)) {
      found = true;
```

Looks like tests are here: https://github.com/iree-org/iree/blob/main/runtime/bindings/python/tests/hal_test.py

> I think this bug only occurs in amd and intel gpus

Strange...

Reply:
I looked at the tests, and dont see where you test the create_device function

Reply:
Sorry yeah, I'd _expect_ tests to be there (or another file in that directory like https://github.com/iree-org/iree/blob/main/runtime/bindings/python/tests/system_api_test.py). We're probably missing test coverage. Ideally any new tests would catch this error (so they would fail before the fix). Do you want to try adding some tests, or shall someone else on the team?

Reply:
Yeah I can create some tests for the create_device function